src/main/java/io/confluent/idesidecar/restapi/proxy/FlinkDataPlaneProxyProcessor.java

```diff
@@ -0,0 +1,90 @@
+package io.confluent.idesidecar.restapi.proxy;
+
+import static io.confluent.idesidecar.restapi.util.SanitizeHeadersUtil.sanitizeHeaders;
+
+import io.confluent.idesidecar.restapi.exceptions.ProcessorFailedException;
+import io.confluent.idesidecar.restapi.processors.Processor;
+import io.confluent.idesidecar.restapi.util.UriUtil;
+import io.vertx.core.Future;
+import io.vertx.core.MultiMap;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.List;
+import java.util.Optional;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+/**
+ * Processor to set the proxy request/response parameters for the data plane.
+ */
+@ApplicationScoped
+public class FlinkDataPlaneProxyProcessor extends Processor<ProxyContext, Future<ProxyContext>> {
+
+  private static final String REGION_HEADER = "x-ccloud-region";
+  private static final String PROVIDER_HEADER = "x-ccloud-provider";
+
+  private final String flinkUrlPattern;
+
+  @ConfigProperty(name = "ide-sidecar.cluster-proxy.http-header-exclusions")
+  List<String> httpHeaderExclusions;
+
+  public FlinkDataPlaneProxyProcessor() {
+    // Default value if config system hasn't initialized the value
+    this.flinkUrlPattern = "https://flink.%s.%s.confluent.cloud";
+  }
+
+  //Allow injection via constructor for production code
+  public FlinkDataPlaneProxyProcessor(
+      @ConfigProperty(name = "ide-sidecar.flink.url-pattern")
+      Optional<String> configuredPattern) {
+    this.flinkUrlPattern = configuredPattern.orElse("https://flink.%s.%s.confluent.cloud");
+  }
```

Comment:
We can do this instead, I'll add another review comment for what to change in the test class.

```suggestion
@ConfigProperty("ide-sidecar.flink.url-pattern")
String flinkUrlPattern;
```